src/lang/base/TypeCache.ml

```diff
@@ -24,22 +24,40 @@ open TypeUtil
 (*                    Library type caching                       *)
 (*****************************************************************)
 
-module StdlibTypeCacher (Q : MakeTEnvFunctor) (R : QualifiedTypes) = struct
+module StdlibTypeCacher
+    (Q : MakeTEnvFunctor)
+    (R : QualifiedTypes)
+    (SR : Rep)
+    (ER : sig
+       type rep
```

Comment:
Can this not be in a new file`ParseUtil.mli` rather than inlined here?

Reply:
Yes - the signature was made explicit because I wasn't sure whether I would need functions that can only exist in TypecheckerERep, but it ended up not being necessary.